File: src/main/java/org/zenbot/szolnok/timetable/batch/batch/step/bus/processor/ActualStopSelectorItemProcessorHelper.java
package org.zenbot.szolnok.timetable.batch.batch.step.bus.processor;

import org.jsoup.nodes.Document;
import org.springframework.stereotype.Component;

@Component
public class ActualStopSelectorItemProcessorHelper {

    private final StringCleaner stringCleaner;

    public ActualStopSelectorItemProcessorHelper(StringCleaner stringCleaner) {
        this.stringCleaner = stringCleaner;
    }

    public String getActualStop(Document htmlDocument, String actualStopSelector) {
        String actualStop = htmlDocument.select(actualStopSelector).text();
        // In the html page they put it insede () signs.
        actualStop = (String) actualStop.subSequence(actualStop.indexOf("(") + 1, actualStop.indexOf(")"));
        if (actualStop.contains("(")) {
            actualStop += ")";
        }
        if (actualStop.endsWith(".")) {
            actualStop = actualStop.substring(0, actualStop.length() - 1);
        }
        return stringCleaner.clean(actualStop);
    }
}

File: szolnok-timetable-batch-bus/src/main/kotlin/org/zenbot/szolnok/timetable/batch/bus/batch/step/bus/processor/ActualStopSelectorItemProcessorHelper.kt
package org.zenbot.szolnok.timetable.batch.bus.batch.step.bus.processor

import org.jsoup.nodes.Document
import org.springframework.stereotype.Component
import org.zenbot.szolnok.timetable.batch.utils.configuration.properties.TimetableSelectorProperties

@Component
class ActualStopSelectorItemProcessorHelper(private val stringCleaner: StringCleaner) {

    fun getActualStop(htmlDocument: Document, selectorProperties: TimetableSelectorProperties): String {
        val actualStopSelector = selectorProperties.actualStopSelector
        var actualStop = htmlDocument.select(actualStopSelector).text()
        // In the html page they put it insede () signs.
        actualStop = actualStop.subSequence(actualStop.indexOf("(") + 1, actualStop.indexOf(")")) as String
        if (actualStop.contains("(")) {
            actualStop += ")"
        }
        if (actualStop.endsWith(".")) {
            actualStop = actualStop.substring(0, actualStop.length - 1)
        }
        return stringCleaner.clean(actualStop)
    }
}
